### PR TITLE
libraries-bom usage docs for end users

### LINKED ISSUE #21: BOM usage docs for End users
i.e. devs who simply import the libraries

### PR DESCRIPTION
**Pull request by elharo**
fixes #21 

Focus on using libraries-bom.

For now this simply publishes the doc on jlbp.dev. We can figure out where to link it from later. 

@garrettjonesgoogle

### CODE REVIEW

File: docs/DECLARING_DEPENDENCIES.md [hunk 1]
```diff
@@ -124,55 +120,6 @@ For an example, see [gax-java#690](https://github.com/googleapis/gax-java/pull/6
 
 See the "Choosing BOMs" section below if you need to import more than one BOM.
 
-## Choosing BOMs
```
Comment (saturnism) [inline docs/DECLARING_DEPENDENCIES.md]: this section has pretty useful information; can we keep?

File: docs/DECLARING_DEPENDENCIES.md [hunk 2]
```diff
@@ -201,10 +148,3 @@ library ecosystem:
 - Each library follows semantic versioning. This means that once a
   library reaches a 1.x version, features and APIs can be added in minor/patch
   releases but not removed within a major version.
-
-This combination of characteristics means that you can generally avoid
```
Comment (saturnism) [inline docs/DECLARING_DEPENDENCIES.md]: this is also valid, any reason for removal?

File: docs/DECLARING_DEPENDENCIES.md (hunk 1 above)
Comment (elharo) [reply docs/DECLARING_DEPENDENCIES.md]: This is from earlier days before we had a working libraries-bom. On of our OKRs is to make libraries-bom the standard means of consuming these libraries. Developers shouldn't need to think about which BOMs to pick when. Just use libraries-bom is a much simpler and easier to grok story. 

It might be worth adding a section about combining BOMs from multiple projects; e.g. libraries-bom and spring BOM, but that's not what this is about and would be a subject for a separate PR.

File: docs/DECLARING_DEPENDENCIES.md (hunk 2 above)
Comment (elharo) [reply docs/DECLARING_DEPENDENCIES.md]: On rereading this, I found it hard to follow and perhaps a bit tautological. #2 needs to go because we want devs to simply import libraries-bom, not multiple BOMs, and #1 needs to go because developers shouldn't need to think about the versions of individual dependencies, just the BOM.

File: docs/DECLARING_DEPENDENCIES.md (hunk 1 above)
Comment (saturnism) [reply docs/DECLARING_DEPENDENCIES.md]: my thought is that there is a difference btwn best practices for java libraries in terms of how to produce and consume boms, vs. preferred method of consumption for GCP client libraries; The latter is in the c.g.c documentation.

It's still super useful to keep the list of the boms, and also adding libraries bom to the list.  Each individual BOM can be used to override 1 component version as a whole. E.g., in situations where gRPC is causing trouble, simply importing a gRPC BOM would solve the issue.

File: docs/DECLARING_DEPENDENCIES.md (hunk 2 above)
Comment (saturnism) [reply docs/DECLARING_DEPENDENCIES.md]: ack; thnx for the info, i feel that makes sense too.

File: docs/DECLARING_DEPENDENCIES.md (hunk 1 above)
Comment (elharo) [reply docs/DECLARING_DEPENDENCIES.md]: That's a very special case though. It does happen. There's an issue in java-storage right now where people are doing that with the http-java-client, but I'm not sure as written this correctly describes that situation. I want to give a very simple message here:

Use libraries-bom.

That perhaps sometimes  specifies more versions than the users technically needs, but in that case no harm done. It is so much easier for users to consume a simple message like this thna a long, convoluted message of all the possible special cases.

Comment (garrettjonesgoogle) [reply docs/DECLARING_DEPENDENCIES.md]: I agree with @saturnism .

We should still include the extended message, but we could also "offload" it to a separate page and provide a link to it, so people who do need those details can get to it.

File: docs/DECLARING_DEPENDENCIES.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+`com.google.cloud:libraries-bom` also covers several other libraries
+that client code does not usually depend on directly. However if you do import
+any of these, use  `com.google.cloud:libraries-bom` to specify their versions
+too:
+
+- [GAX Google API Extensions for Java](https://github.com/googleapis/gax-java)
+- [Google Auth Library](https://github.com/googleapis/google-auth-library-java)
+- [Google API Client Library for Java](https://github.com/googleapis/google-api-java-client)
+- [Grafeas Artifact Metadata API](https://github.com/grafeas/grafeas)
```
Comment (suztomo) [inline docs/DECLARING_DEPENDENCIES.md]:
```suggestion
- [Grafeas Artifact Metadata API](https://github.com/googleapis/java-grafeas)
```

as per https://search.maven.org/artifact/io.grafeas/grafeas/0.24.0/jar

File: docs/DECLARING_DEPENDENCIES.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+dependency tree.
+
+When a library publishes multiple artifacts, using different versions
+for different artifacts may cause dependency conflicts. To fix this
+problem, you can import a BOM to force consistent versions, as long as
+you are using at least Gradle 4.6. To do this:
+
+- Turn on BOM support:
+  - If you are using Gradle 4.x:
+
```
Comment (suztomo) [inline docs/DECLARING_DEPENDENCIES.md]: Case of Gradle 4.x is missing.

File: docs/DECLARING_DEPENDENCIES.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+- [Grafeas Artifact Metadata API](https://github.com/grafeas/grafeas)
+
+## Ensuring Compatibility
+
+To ensure that your own project uses compatible versions of these
+libraries, follow the guidance below for your build system.
+
+### Maven
+
+#### Use the requireUpperBoundDeps enforcer rule
```
Comment (suztomo) [inline docs/DECLARING_DEPENDENCIES.md]: Shall we put "Import BOMs to ensure dependencies are consistent" first, because the purpose of this document is to encourage users to import the BOM.

File: docs/DECLARING_DEPENDENCIES.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+```
+
+When the newest versions are not compatible, you can use BOMs to
+select compatible versions.
+
+#### Import BOMs to ensure dependencies are consistent
```
Comment (suztomo) [inline docs/DECLARING_DEPENDENCIES.md]:
```suggestion
#### Import GCP Libraries BOM to ensure dependencies are consistent
```

This section "Ensuring Compatibility" is written for generic usage of BOMs (not specific to libraries-bom), while this document is "Declaring Dependencies on Google Cloud Java Libraries". Is this intentional?

File: docs/DECLARING_DEPENDENCIES.md (hunk 3 above)
Comment (elharo) [reply docs/DECLARING_DEPENDENCIES.md]: done

File: docs/DECLARING_DEPENDENCIES.md (hunk 6 above)
Comment (elharo) [reply docs/DECLARING_DEPENDENCIES.md]: done

File: docs/DECLARING_DEPENDENCIES.md (hunk 5 above)
Comment (elharo) [reply docs/DECLARING_DEPENDENCIES.md]: done

File: docs/DECLARING_DEPENDENCIES.md (hunk 6 above)
Comment (elharo) [reply docs/DECLARING_DEPENDENCIES.md]: done

File: docs/DECLARING_DEPENDENCIES.md (hunk 4 above)
Comment (elharo) [reply docs/DECLARING_DEPENDENCIES.md]: done
